### ISSUE TITLE
MA0018 is reported when static members in generic type depends on generic parameter

### ISSUE THREAD

**Issue by hankovich** [meziantou/Meziantou.Analyzer#565]
### Version of the Meziantou.Analyzer NuGet package

2.0.68

### Rule Identifier

MA0018

### Target Framework

net8.0

### C# Language version

C# 12

### Description

It should be ok to have static members that depend on generic parameters

### Reproduction Steps

Minimal code:

```c#
public static class Container<T>
{
    public static T? Value { get; set; }
}
```

**Comment by meziantou** [comment]
I think the solution is to deprecate the rule in favor of CA1000. This rule has been added when there was no built-in analyzers and FxCop was almost dead. Nowadays there is no need to have duplicated rules.
